libraries/AP_RangeFinder/AP_RangeFinder_NMEA.h

```diff
@@ -0,0 +1,77 @@
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "RangeFinder.h"
+#include "RangeFinder_Backend.h"
+
+class AP_RangeFinder_NMEA : public AP_RangeFinder_Backend
+{
+
+public:
+    // constructor
+    AP_RangeFinder_NMEA(RangeFinder::RangeFinder_State &_state,
+                        AP_SerialManager &serial_manager,
+                        uint8_t serial_instance);
+
+    // static detection function
+    static bool detect(AP_SerialManager &serial_manager, uint8_t serial_instance);
+
+    // update state
+    void update(void) override;
+
+protected:
+
+    virtual MAV_DISTANCE_SENSOR _get_mav_distance_sensor_type() const override {
+        return MAV_DISTANCE_SENSOR_ULTRASOUND;
+    }
+
+private:
+
+    /// enum for handled messages
+    enum _sentence_types {      //there are some more than 10 fields in some sentences , thus we have to increase these value.
```

Comment:
The comment seems to be in the wrong place?

Also, adding `: uint8_t` might be worth it.

Reply:
txs, added uint8_t